CMakeLists.txt

```diff
@@ -0,0 +1,86 @@
+cmake_minimum_required(VERSION 3.16)
+project(tt-metal)
+
+set(CMAKE_CXX_FLAGS_RELEASE "-O3")
+set(CMAKE_CXX_FLAGS_DEBUG "-O0 -g -DDEBUG=DEBUG")
+set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O3 -g -DDEBUG=DEBUG")
+set(CMAKE_CXX_FLAGS_CI "-O3 -DDEBUG=DEBUG")   # idk if this is the right way to add a custom build type but it works.. also removed --verbose linker option
```

Comment:
I never never understood why and didn't observe CMAKE_BUILD_TYPE not working in my vscode. I was doing it that way before but I went the CMAKE_BUILD_TYPE route since it's cleaner imo and generator statements are ugly/not that readable imo

If we need to switch over in the future it would pre simple, just add it to the `compiler_flags` intf lib and have a CONFIG variable instead of CMAKE_BUILD_TYPE

Reply:
i agree with using generator statements more